src/main/java/com/google/devtools/build/lib/skyframe/TreeArtifactValue.java

```diff
@@ -70,13 +72,16 @@ public PathFragment apply(Artifact artifact) {
   static TreeArtifactValue create(Map<TreeFileArtifact, FileArtifactValue> childFileValues) {
     Map<String, FileArtifactValue> digestBuilder =
         Maps.newHashMapWithExpectedSize(childFileValues.size());
+    boolean isRemote = true;
     for (Map.Entry<TreeFileArtifact, FileArtifactValue> e : childFileValues.entrySet()) {
+      isRemote = isRemote && e.getValue().isRemote();
```

Comment:
it seems like things get strange when some files within the tree are remote and others non-remote. can you add some comments about when that would happen and the expected semantics?

Reply:
Can that happen? A TreeArtifact can only ever be created by one piece by an action. 

Can it maybe happen that only parts of it are materialized?

Reply:
Sure, I ll add some comments (and checks maybe). I think it should be either remote or local and not a mix of both.

Reply:
done